### PR TITLE
Makes supermatter give some light advice on how to fix it when it delams

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

It just gives the basic advice, and only when pertinent: add nitrous if it gets hot, turn off the FRIGGIN EMITTERS, try to keep the N2 percent high. Doesn't say anything about piping or, like, how to do any of these.

## Why It's Good For The Game

i keep seeing people get into the CE hardsuit to use a fire extinguisher on the supermatter while the emitters are still running even though that's just banging your head against a wall repeatedly to bust through it when there's an open door next to you and it's officially made me mad

## Changelog
:cl:
add: Supermatter now bugs you about how to fix it
/:cl:

### CODE REVIEW

**Comment by MrJWhit** [comment]
Hugbox.

**Comment by Solaris-Shade** [comment]
I'm with MrWhit on this. These voice prompts take the educating the newbies out of the player's hands.

 I feel that there needs to be a respectable level of danger for all engines in SS13.

An alternative to the PA prompts is to update the in-game engineer book with some troubleshooting tips at the end of the book.

**Comment by Putnam3145** [comment]
> 
> 
> I'm with MrWhit on this. These voice prompts take the educating the newbies out of the player's hands.
> 
> I feel that there needs to be a respectable level of danger for all engines in SS13.
> 
> An alternative to the PA prompts is to update the in-game engineer book with some troubleshooting tips at the end of the book.

the engineering book, as far as i know, includes the supermatter TG wiki article, which includes everything I mention in this PR. the problem is that clearly nobody wants to read it

**Comment by BlueWildrose** [comment]
I don't get why we need to make SS13 solely a "wiki game". I feel like when you have to depend on the wiki for any form of guidance in any game, it detracts from the experience and honestly feels like people are trying to 'gatekeep' by requiring that players take more time out of their way to look something up, or perhaps seeking guidance from others. Minecraft used to be solely wiki only, but then they have the Recipe book in the vanilla game now, and if you're playing mods, there's JEI and other tools used to guide you in-game in a properly set up mod. These I feel are the step in the right direction for that game.

In SS13, nowadays there's various guides that tell you what to do what in-game, like the operating table computer telling you the next step needed in a surgery, the examine text of machines telling you how efficient they are without needing to use an RPED to find out such facts, or the examine text of a window, wall, or machine frame giving you subtle hints on what's needed to be done next with it. I don't see why telling people about why a supermatter is dying would be considered a problem. It's arguably more difficult to harness than the singularity/tesla, and there's very little in-game documentation about how it works - you have to rely solely on the wiki for knowing how it works, or having someone tell you how it works in-game. The problem with that latter part is that veterans eventually just quit or start being lazy and can't be assed to talk to new players when they'd rather just lay down in the garden growing melons or whatever the fuck it is they do nowadays.

If nothing else, having this explained to new users in a more precise way means the game itself can tell them when they're being stupid, and it doesn't depend on the possibility that they're going to have a toxic veteran player on a bad day who's ready to toolbox someone at a moment's notice for even committing the slightest slip-up (_this is a genuine problem that I've seen that happened several times this year that really needs to stop._) who spends more time telling them that they're an idiot more than telling them what they're doing wrong. 

I understand that part of SS13's experience is the mystery and discovery of hidden or important things, and that's been a few servers like Goon and Lifeweb's strongest intrigues, but that's because they're closed source. Here, we're open source. I think it's okay to give hints like this. There is so many mechanics to this game that have to be memorized for you to even be good at them, and for some it will take _years_ before they could understand every aspect to the game. Changes like this will be beneficial to new players and to ensure that they stay. Wouldn't you prefer that we have players, especially since we're currently going through a lowpop funk nowadays?

**Comment by ghost** [comment]
Leave it as a book and put it in a plainly obvious area. If they wanna learn and figure it out with a wiki or book then it's there, but having it spout out easy ways to fix it just seems a bit too hugbox

**Comment by silicons** [comment]
this is a good PR
`Warning: high power detected in the supermatter. Possible causes include emitters being online, a power-rich fuel mix or high levels of carbon dioxide inhibiting power loss. Turn off emitters and perhaps add more nitrogen.`

i fail to see anything "hugbox" about this
these are basic things that anyone doing engineering should know how to do
sure, it'd be nice if people read the wiki or the book (why the fuck aren't they) but literally who is it hurting by adding this?

the only reason i'd personally have to oppose this is the ic immersion of the monitor telling you what to do and the potential ramifications of that, rather than "this is too hugbox". it's not. if you wanted to actually sabotage the engine, this won't suddenly teach everyone how to fix intentional fuckery.

edit: to clarify i think it's weird to have this come out as some sort of error message/tip of the day from the warning system. i don't think it's bad enough to justify opposing this though.

**Comment by LetterN** [comment]
Hugbox. Would be better if you moved this into tips.txt instead.

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -730,10 +732,15 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
-					radio.talk_into(src, "DANGER: CHARGE INERTIA CHAIN REACTION IN PROGRESS.", engineering_channel)
+			if(removed.return_temperature() > TEMPERATURE_HEURISTIC_THRESHOLD)
+				radio.talk_into(src, "High temperatures detected. Recommend adding nitrous oxide or pluoxium to the chamber to reduce integrity loss.")
+			switch(power)
+				if(POWER_HEURISTIC_THRESHOLD to POWER_PENALTY_THRESHOLD)
+					radio.talk_into(src, "Warning: high power detected in the supermatter. Possible causes include emitters being online, a power-rich fuel mix or high levels of carbon dioxide inhibiting power loss. Turn off emitters and perhaps add more nitrogen.")
+				if(POWER_PENALTY_THRESHOLD to INFINITY)
+					radio.talk_into(src, "Warning: Hyperstructure has reached dangerous power level.", engineering_channel)
+					if(powerloss_inhibitor < 0.5)
+						radio.talk_into(src, "DANGER: CHARGE INERTIA CHAIN REACTION IN PROGRESS.", engineering_channel)
```
Comment (LetterN) [inline code/modules/power/supermatter/supermatter.dm:743]:
```suggestion
			if(removed.return_temperature() > TEMPERATURE_HEURISTIC_THRESHOLD)
				radio.talk_into(src, "High temperatures detected. Recommend adding nitrous oxide or pluoxium to the chamber to reduce integrity loss.", engineering_channel)
			switch(power)
				if(POWER_HEURISTIC_THRESHOLD to POWER_PENALTY_THRESHOLD)
					radio.talk_into(src, "Warning: high power detected in the supermatter. Possible causes include emitters being online, a power-rich fuel mix or high levels of carbon dioxide inhibiting power loss. Turn off emitters and perhaps add more nitrogen.", engineering_channel)
				if(POWER_PENALTY_THRESHOLD to INFINITY)
					radio.talk_into(src, "Warning: Hyperstructure has reached dangerous power level.", engineering_channel)
					if(powerloss_inhibitor < 0.5)
						radio.talk_into(src, "DANGER: CHARGE INERTIA CHAIN REACTION IN PROGRESS.", engineering_channel)
```

**Comment by MrJWhit** [comment]
It's hugboxing because nothing else in ss13 will tell you explicitly how to fix it. Sometimes people mess up, and that's ok, it's a game about learning. The SM blows up, they ask what they did wrong, and they got told that their emitters were on. Someone not wanting to learn isn't a code issue, it's quite literally a player issue for not wanting to learn or ask questions.

_IF_ we did want to go around this, why does high temperature specify **pluox**, something you generally need to work to make and isn't a casual additive? Why doesn't the hot warning not specify the fact that the chamber could be on fire? Why isn't there anything specifying pressure?

**Comment by Linzolle** [comment]
I'm not against it giving warnings, but I don't like how specific it is listing specific gasses instead of saying something generic like "coolant"
I feel like it should give a little nudge as to how to fix, but not tell it outright? It feels strange to me.

**Comment by shellspeed1** [comment]
Idea here
What if we added these tips as portions of the ntsims program. From there we could make the engine say check ntsims for more details. Traitors could emag the console and rename the brick a pc program while also killing the tcomms wifi server for their delams so that people don't get hints for them.
Edit: hugbox

**Comment by Arturlang** [comment]
These are basic as fuck suggestions, a traitor wouldnt specifically try to deny people this due to how simple they are. This game's skill requirement is really high, and it's all the better if we reduce it

**Comment by keronshb** [comment]
SM's already hugboxed to hell.

Fires tend to put themselves out even with basic cooling.

SM surge is a joke of an event.

You can't even shoot the SM to death anymore.

**Comment by Putnam3145** [comment]
> 
> 
> SM's already hugboxed to hell.
> 
> Fires tend to put themselves out even with basic cooling.

Yes, this is actually good, turning off the emitters should prevent the supermatter's death even with the roundstart setup if it's a "normal fire". However, I've seen enough delams in the last week or so to know that, yes, it *delaminates anyway*, so I'm thinking maybe you're just, like, wrong? About this?

> SM surge is a joke of an event.

It's strong enough to make any and all CO2 setups actually outright explode without grounding rods. I don't know how to make it stronger without making it impossibly strong.

> You can't even shoot the SM to death anymore.

I've been thinking about buffing the effects of bullets on the supermatter, yes, but if you think "you can bypass the entirety of the supermatter's design and all ability for the engineers to do fuck all about dealing with it via a .357 and two seconds" was good design, sorry, can't say I agree.

**Comment by Ragolution** [comment]
Seems like a skill issue.

Would be funny/good if the CE hardsuit specifically was what told people how to fiddle with the Supermatter, though.
Maybe a special module on the Chief Engineer's PDA cart? The supermatter monitoring one already exists so you've got good framework there.

**Comment by RummyRin** [comment]
In all fairness Extinguishing an SM fire is stupid viable for cooling and helping gas flow.

**Comment by Ragolution** [comment]
I'm just gunna reinforce my suggestion that this be made part of the CE's hardsuit. Having it nag over comms seems like it might be spammy if the right conditions are met, nevermind annoying. Even then, at that point, you'd be encouraging an overreliance on the system, resulting in Engineers never visiting Engineering and simply relying on comms. Something that kind of already happened/happens. Making a "OH GOD IT'S DELAMMING" 101 textbook (not the wiki page) might be helpful. Maybe compose a short pamphlet on ways to try to fix it.

On the other hand, having it be a caveat for the CE hardsuit makes me go :). Anything that makes people want to wear it less.

**Comment by DeltaFire15** [comment]
This gives a considerable amount of information via comms without engineers even needing to check the monitoring programs, aswell as having the potential of being quite spammy / textwalling provided certain conditions are met.
Besides those though, the tips it itself gives are quite substantial and could be handled via a simple "supermatter meltdowns 101" book, not current-situation-based comms readouts which do fit the term "hugbox".
Closing this PR due to these.

**Pull request closed without merging**